src/test/kotlin/com/example/iot_management_device/service/device/DeviceServiceImplTest.kt

```diff
@@ -0,0 +1,248 @@
+package com.example.iot_management_device.service.device
+
+import com.example.iot_management_device.dto.device.request.DeviceRequestDto
+import com.example.iot_management_device.dto.device.response.DeviceResponseDto
+import com.example.iot_management_device.dto.device.request.DeviceUpdateRequestDto
+import com.example.iot_management_device.exception.EntityNotFoundException
+import com.example.iot_management_device.mapper.DeviceMapper
+import com.example.iot_management_device.model.Device
+import com.example.iot_management_device.model.DeviceStatusType
+import com.example.iot_management_device.repository.DeviceRepository
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.verify
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.assertThrows
+import java.util.*
+
+class DeviceServiceImplTest {
+
+    @MockK
+    private lateinit var deviceRepository: DeviceRepository
+
+    @MockK
+    private lateinit var deviceMapper: DeviceMapper
+
+    @InjectMockKs
+    lateinit var deviceService: DeviceServiceImpl
+
+    private lateinit var deviceRequestDto: DeviceRequestDto
+    private lateinit var deviceUpdateDto: DeviceUpdateRequestDto
+    private lateinit var device: Device
+    private lateinit var savedDevice: Device
+    private lateinit var deviceResponseDto: DeviceResponseDto
+
+    @BeforeEach
+    fun setUp() {
+        MockKAnnotations.init(this)
+
+        deviceRequestDto = DeviceRequestDto(
```

Comment:
You can create DeviceFixtureClass and not repeat yourself

Reply:
Done